src/renderers/dom/ReactDOM.js

```diff
@@ -133,6 +133,13 @@ if (__DEV__) {
       }
     }
   }
+
+  // Check if using Polymer and we are not using native shadow
+  var shadyDom = window.Polymer !== undefined && !window.Polymer.useNativeShadow;
```

Comment:
Should we emit the warning if `Polymer.useShadow` is false? I'm not sure what it means (will need to read some docs) but from the code it seems like `useNativeShadow` would then also become `false` regardless of the polyfill.